apps/spruce/src/pages/task/taskTabs/TaskHistory/TestFailureSearchInput/index.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useMemo, useState } from "react";
+import styled from "@emotion/styled";
+import TextInput from "@leafygreen-ui/text-input";
+import debounce from "lodash.debounce";
+import { size } from "@evg-ui/lib/constants/tokens";
+import { useQueryParam } from "hooks/useQueryParam";
+import { TaskHistoryOptions } from "../types";
+
+interface Props {
+  numMatchingResults: number;
+}
+export const TestFailureSearchInput: React.FC<Props> = ({
+  numMatchingResults,
+}) => {
+  const [failingTest, setFailingTest] = useQueryParam<string>(
+    TaskHistoryOptions.FailingTest,
+    "",
+  );
+
+  const [searchTerm, setSearchTerm] = useState(failingTest);
+  const updateQueryParamWithDebounce = useMemo(
+    () => debounce((str) => setFailingTest(str ? str : undefined), 150),
```

Comment:
We seem to debounce with a random number for every single debounce input. I think we should settle on something and make it consistent across the app. 
Can you make this change across the 3 usages of debounce in spruce?